### DISCUSSION TITLE
Don't throw exception in serialize/deserialize macros when we have no some field

### DISCUSSION THREAD

**Discussion by keygang** [nlohmann/json#2395] [Q&A] [answered]
Is it some possibility parse json using `NLOHMANN DEFINE_TYPE_NON INTRUSIVE` with some missed fields?

```
#include <iostream>
#include <nlohmann/json.hpp>

using namespace nlohmann;

struct Lol {
  struct Izi {
    int lol = 0;
    int kek = 0;
  };

  Izi izi;
};

NLOHMANN_DEFINE_TYPE_NON_INTRUSIVE(Lol::Izi, lol, kek);
NLOHMANN_DEFINE_TYPE_NON_INTRUSIVE(Lol, izi);

int main() {
  json j = {{"izi", {{"lol", 228}}}};

  Lol lol;
  nlohmann::adl_serializer<Lol>::from_json(j, lol);
  std::cout << "kek: " << lol.izi.kek << ";  lol: " << lol.izi.lol;

  return 0;
}
```
If i run it i get 
`libc++abi.dylib: terminating with uncaught exception of type nlohmann::detail::out_of_range: [json.exception.out_of_range.403] key 'kek' not found`.
And i want to exit without any errors, and get the next output: `kek: 0; lol: 228`.

**Comment by nlohmann** [comment, marked as answer]
If you want a different behavior than what the macro offers, you need to define your own from_json function.
